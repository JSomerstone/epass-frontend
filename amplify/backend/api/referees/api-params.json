{
    "paths": [
        {
            "name": "/referees",
            "lambdaFunction": "refereesApi",
            "privacy": {
                "private": true,
                "auth": [
                    "/POST",
                    "/GET",
                    "/PUT",
                    "/PATCH"
                ]
            }
        },
        {
            "name": "/tournaments/{year}/{id}",
            "lambdaFunction": "tournamentsApi",
            "privacy": {
                "private": true,
                "auth": [
                    "/POST",
                    "/GET",
                    "/PUT",
                    "/PATCH"
                ]
            }
        }
    ],
    "resourceName": "referees",
    "apiName": "referees",
    "functionArns": [
        {
            "lambdaFunction": "refereesApi"
        },
        {
            "lambdaFunction": "tournamentsApi"
        }
    ],
    "privacy": {
        "auth": 2,
        "unauth": 0,
        "authRoleName": "amplify-epass-dev-130844-authRole",
        "unAuthRoleName": "amplify-epass-dev-130844-unauthRole"
    },
    "dependsOn": [
        {
            "category": "function",
            "resourceName": "refereesApi",
            "attributes": [
                "Name",
                "Arn"
            ]
        },
        {
            "category": "function",
            "resourceName": "tournamentsApi",
            "attributes": [
                "Name",
                "Arn"
            ]
        }
    ]
}