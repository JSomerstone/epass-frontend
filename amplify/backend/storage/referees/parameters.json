{
    "tableName": "referees",
    "partitionKeyName": "id",
    "partitionKeyType": "S",
    "sortKeyName": "userId",
    "sortKeyType": "S"
}