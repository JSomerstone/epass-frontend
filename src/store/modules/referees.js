import Referee from "../models/RefereeClass";
import Association from "../models/Association";
import { successMessage, notifyException, warningMessage } from "../../utils/notificationUtils";
import { createReferee, updateReferee, createAssociation, updateAssociation } from "../../graphql/mutations";
import { listReferees, getReferee, listAssociations } from "../../graphql/queries";
import { API } from "aws-amplify";
const courseConductors = require("../../assets/course-conductors.json");

const state = {
  loading: 0,
  referees: [],
  queryResult: [],
  current: {},
  conductors: courseConductors,
  associations: [],
  debug: false,
};

const mutationTypes = {
  SET_LOADING: "set-loading",
  SET_REFEREES: "set-referees",
  ADD_REFEREE: "add-referee",
  UPDATE_REFEREE: "update-referee",
  SET_CURRENT: "set-current",
  SET_ASSOCIATIONS: "set-associations",
  ADD_ASSOCIATION: "add-association",
  UPDATE_ASSOCIATION: "update-association",
  QUERY_RESULT: "set-query-result",
};

const mutations = {
  [mutationTypes.SET_LOADING](state, loading) {
    state.loading = state.loading + (loading ? 1 : -1);
    if (state.loading < 0) {
      state.loading = 0;
    }
  },
  [mutationTypes.SET_REFEREES](state, referees) {
    state.referees = referees;
  },
  [mutationTypes.SET_ASSOCIATIONS](state, list) {
    state.associations = list;
  },
  [mutationTypes.ADD_ASSOCIATION](state, item) {
    state.associations.push(item);
  },
  [mutationTypes.UPDATE_ASSOCIATION](state, item) {
    const index = state.associations.find(r => r.id == item.id);
    state.associations[index] = item;
  },
  [mutationTypes.ADD_REFEREE](state, referee) {
    state.referees.push(referee);
  },
  [mutationTypes.UPDATE_REFEREE](state, referee) {
    const index = state.referees.find(r => r.id == referee.id);
    state.referees[index] = referee;
  },
  [mutationTypes.SET_CURRENT](state, referee) {
    state.current = new Referee(referee);
  },
  [mutationTypes.QUERY_RESULT](state, refereeList) {
    state.queryResult = refereeList;
  },
};

const actions = {
  setLoading({ commit }, { loading }) {
    commit(mutationTypes.SET_LOADING, loading);
  },

  setCurrent: async ({ commit }, { id }) => {
    API.graphql({
      query: getReferee,
      variables: { id },
    })
      .then(
        result => {
          const referee = result.data.getReferee;
          commit(mutationTypes.SET_CURRENT, referee);
        }
      )
      .catch(notifyException)
      .finally(() => {
        commit(mutationTypes.SET_LOADING, false);
      });
  },
  load: async ({ commit }) => {
    commit(mutationTypes.SET_LOADING, true);
    API.graphql({
      query: listReferees,
    }).then(
      result => {
        const referees = result.data.listReferees.items;
        commit(mutationTypes.SET_REFEREES, referees);
      }
    )
      .catch(notifyException)
      .finally( () => { commit(mutationTypes.SET_LOADING, false); }
    );
  },
  loadCurrent: async (
    { commit, rootGetters },
    { onSuccess = () => { }, onFailure = () => { } }
  ) => {
    const { userId = null } = rootGetters["auth/user"];
    if (null == userId) {
      warningMessage("Unable to load user profile");
      onFailure();
      return;
    }
    commit(mutationTypes.setLoading, true);
    API.graphql({
      query: listReferees,
      variables: {
        filter: {
          userId: { eq: userId },
        },
      },
    }).then(
      result => {
        const { items } = result.data.listReferees;
        console.log({ items });
        if (items.length == 0) {
          onFailure();
        } else {
          commit(mutationTypes.SET_CURRENT, items[0]);
          onSuccess(items[0]);
        }
      }
    )
      .catch(notifyException)
      .finally(() => {
        commit(mutationTypes.setLoading, false)
      });
  },
  create: async ({ commit, dispatch }, { referee, onSuccess = () => {} }) => {
    commit(mutationTypes.SET_LOADING, true);
    delete referee.clinic;
    try {
      const result = await API.graphql({
        query: createReferee,
        variables: { input: referee },
      });
      commit(mutationTypes.ADD_REFEREE, result.data.createReferee);
      successMessage("Referee added");
      onSuccess(result.data.createReferee);
      dispatch("load");
    } catch (err) {
      notifyException(err);
    }
    commit(mutationTypes.SET_LOADING, false);
  },
  async update({ commit, dispatch }, { referee, onSuccess = () => {} }) {
    try {
      const result = await API.graphql({
        query: updateReferee,
        variables: {
          input: referee,
        },
      });
      commit(mutationTypes.UPDATE_REFEREE, result.data.updateReferee);
      onSuccess(result.data.updateReferee);
      successMessage("Updated");
      dispatch('setCurrent', result.data.updateReferee);
    } catch (err) {
      notifyException(err);
    }
  },
  async findByEmail({ commit }, { email, onSuccess = () => {} }) {
    commit(mutationTypes.SET_LOADING, true);
    API.graphql({
      query: listReferees,
      variables: {
        filter: {
          email: { eq: email },
        },
      },
    })
      .then((result) => {
        const referees = result.data.listReferees.items.map((r) => {
          return { ...r };
        });
        onSuccess(referees);
        commit(mutationTypes.QUERY_RESULT, referees);
      })
      .catch(notifyException)
      .finally(() => {
        commit(mutationTypes.SET_LOADING, false);
      });
  },
  async loadAssociations({ commit }) {
    commit(mutationTypes.SET_LOADING, true);
    API.graphql({
      query: listAssociations
    }).then(result => {
      commit(mutationTypes.SET_ASSOCIATIONS, result.data.listAssociations.items)
    }).catch(notifyException)
      .finally(() => {
        commit(mutationTypes.SET_LOADING, false);
    })
  },
  async createAssociation({ commit }, { association, onSuccess = () => { } }) {
    commit(mutationTypes.SET_LOADING, true);
    delete association.id;
    API.graphql({
        query: createAssociation,
        variables: {  input: new Association(association) }
    }).then(result => {
      onSuccess(result.data.createAssociation);
      commit(mutationTypes.ADD_ASSOCIATION, result.data.createAssociation);
    }).catch(notifyException)
    .finally(() => {
      commit(mutationTypes.SET_LOADING, false);
    });
  },
  async updateAssociation({ commit }, { association, onSuccess = () => { } }) {
    commit(mutationTypes.SET_LOADING, true);
    API.graphql({
        query: updateAssociation,
        variables: { input: new Association(association) }
      }).then(result => {
        onSuccess(result.data.updateAssociation);
        commit(mutationTypes.UPDATE_ASSOCIATION, result.data.updateAssociation);
      })
      .catch(notifyException)
      .finally(() => {
        commit(mutationTypes.SET_LOADING, false);
      });
  },
  reset({ commit }) {
    commit(mutationTypes.SET_REFEREES, []);
    commit(mutationTypes.SET_CURRENT, {});
    commit(mutationTypes.SET_ASSOCIATIONS, []);
  }
};

const referees = {
  namespaced: true,
  state,
  getters: {
    debug: (state) => state.debug,
    loading: (state) => state.loading > 0,
    all: (state) => state.referees,
    current: (state) => state.current,
    search: (state) => (query) => {
      const all = state.referees;
      return query == ""
        ? all
        : all
            .filter((ref) => {
              const { firstName, lastName, country, email = "" } = ref;
              const match = [firstName, lastName, country, email].find((prop) =>
                prop.toLowerCase().includes(query.toString().toLowerCase())
              );
              return Boolean(match);
            })
            .sort((r0, r1) => r0.firstName > r1.firstName);
    },
    byId: (state) => (id) => {
      return state.referees.find((r) => r.id === id);
    },

    byUserId: (state) => (userId) => {
      return state.referees.find((r) => r.userId === userId);
    },
    conductors: state => state.conductors,
    nationalAssociations: state => state.associations,
    association: (state) => (id) => {
      return state.associations.find((r) => r.id === id);
    },
    associationCountry: (state) => (country) => {
      return state.associations.filter(
        a => a.country.toLowerCase().includes(country.toLowerCase())
      );
    }
  },
  mutations,
  actions,
};
export default referees;
