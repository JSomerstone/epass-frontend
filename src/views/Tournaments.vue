<template>
  <layout>
    <template v-slot:title>Tournaments {{ year }}</template>
    <tournament-form class="tournament-form"></tournament-form>
    <tournament-list :loading="isLoading" :items="tournaments" />
  </layout>
</template>
<style lang="css">
  .tournament-form {
    margin-bottom: 3rem;
  }
</style>
<script>
import Layout from "../components/Layout"
import TournamentList from "../components/TournamentList";
import TournamentForm from "../components/TournamentForm";

export default {
  components: {
    Layout,
    TournamentList,
    TournamentForm
  },
  data() {
    return {
      year:  this.$route.params.year || new Date().getFullYear(),
    }
  },
  computed: {
    tournaments: function() {
      return this.$store.getters['tournaments/all'] || [];
    },
    isLoading: function() {
      return this.$store.getters['tournaments/loading'];
    }
  },
  methods: {
    fetchData: function() {
      this.$store.dispatch("tournaments/load", { year: this.year });
      this.$store.dispatch("tournaments/loadTeams");
      this.$store.dispatch("referees/load");
    }
  },
  created () {
    // fetch the data when the view is created and the data is
    // already being observed
    this.fetchData()
  },
  watch: {
    // call again the method if the route changes
    '$route': 'fetchData'
  },
}
</script>