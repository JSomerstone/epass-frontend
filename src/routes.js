import About from "./views/About.vue";
import ErrorPage from "./views/ErrorPage.vue";
import TournamentPage from "./views/TournamentPage.vue";
import Tournaments from "./views/Tournaments.vue";
import Signup from "./views/Signup.vue";
import Settings from "./views/Settings.vue";
import Statistics from "./views/Statistics.vue";
import PrintEpass from "./views/PrintEpass.vue";
import Privacy from "./views/Privacy.vue";
import ForgotPassword from "./views/ForgotPassword.vue";

export default [
  {
    path: "/",
    name: "frontpage",
    meta: { title: "Frontpage" },
    component: Tournaments,
  },
  {
    path: "/about/",
    name: "about",
    component: About,
  },
  {
    path: "/settings/:category?",
    name: "settings",
    component: Settings,
  },
  {
    path: "/tournaments/:year/",
    name: "tournaments",
    component: Tournaments,
    meta: { title: "Tournaments" },
  },
  {
    path: "/statistics/:year?",
    name: "statistics",
    component: Statistics,
  },
  {
    path: "/epass/:refereeId/:year?",
    name: "print-epass",
    component: PrintEpass,
  },
  {
    path: "/tournament/:id/",
    name: "tournament",
    component: TournamentPage,
  },
  {
    path: "/signup/",
    name: "signup",
    component: Signup,
    props: { tab: 1 },
  },
  {
    path: "/login/",
    name: "login",
    component: Signup,
    props: { tab: 0 },
  },
  {
    path: "/privacy",
    name: "privacy",
    component: Privacy,
    meta: { title: "Privacy Policy" },
  },
  {
    path: "/forgot-password/",
    name: "forgot-password",
    component: ForgotPassword,
  },
  {
    path: "*",
    props: true,
    name: "error",
    component: ErrorPage,
  },
]