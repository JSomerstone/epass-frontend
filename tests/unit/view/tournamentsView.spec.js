import { shallowMount, mount } from "@vue/test-utils";
import Vuex from "vuex";
import localVue from "../localVue";
import Tournaments from "../../../src/views/Tournaments.vue";
import tournamentModule from "../../../src/store/modules/tournaments";
import refereeModule from "../../../src/store/modules/referees";
import VueRouter from "vue-router";
import routes from "@/routes";
import Referee from "@/store/models/RefereeClass";
import referees from "../../data/referee.json";
import stubs from "../../stubs";

describe("Tournaments-view", () => {
  let state, store, route;
  let router = new VueRouter({ routes });
  route = {
    params: { year: 2020 },
    query: { open: true }
  }
  state = {
    tournaments: tournamentModule.state,
    referees: refereeModule.state,
  };
  beforeEach(() => {
    let storeProps = {
      modules: {
        tournaments: {
          state: state.tournaments,
          namespaced: true,
          getters: {
            all: () => [],
            loading: () => false,
            debug: () => false,
            filter: () => state.tournaments.filter,
            teams: () => [],
          },
          actions: {
            load: jest.fn(),
            loadTeams: jest.fn(),
            addTeam: jest.fn(),
          }
        },
        referees: {
          state: state.referees,
          namespaced: true,
          getters: {
            current: () => new Referee(referees[0]),
            all: () => referees,
            search: () => jest.fn(),
          },
          actions: {
            load: jest.fn()
          }
        },
      },
    };
    store = new Vuex.Store(storeProps);
  });

  it("renders correctly with empty list", () => {
    const wrapper = shallowMount(Tournaments, {
      store,
      router,
      localVue,
      stubs: ["router-link", "b-button"],
      mocks: {
        $route: route
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("renders complete view when mounted", () => {
    const wrapper = mount(Tournaments, {
      store,
      router,
      localVue,
      stubs,
      mocks: {
        $route: route,
        $buefy: jest.fn(),
      }
    });
    expect(wrapper).toMatchSnapshot();

  });
});