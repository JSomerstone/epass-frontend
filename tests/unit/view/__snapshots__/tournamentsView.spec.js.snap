// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tournaments-view renders complete view when mounted 1`] = `
<div>
  <section class="header">
    <b-navbar-stub closeonclick="true" mobileburger="true"><template></template> <template></template> <template></template></b-navbar-stub>
  </section>
  <section class="hero">
    <div class="hero-body">
      <div class="container">
        <h1 class="title">Tournaments</h1>
        <h2 class="subtitle">2020</h2>
      </div>
    </div>
  </section>
  <section class="content">
    <b-loading-stub isfullpage="true" animation="fade" oncancel="function _default() {}"></b-loading-stub>
    <div class="container">
      <h1></h1>
      <b-collapse-stub animation="slide" ariaid="newEntryForm" position="is-top" class="card tournament-form">
        <div role="button" class="card-header"><span class="card-header-title"><span>Add new tournament</span></span> <a class="card-header-icon">
            <b-icon-stub icon="menu-up"></b-icon-stub>
          </a></div>
        <div class="card-content">
          <div class="columns">
            <div class="column is-half">
              <div class="field">
                <div class="columns">
                  <div class="column is-half">
                    <b-field-stub label="Name of tournament" addons="true" labelposition="on-border">
                      <b-input-stub expanded="true" usehtml5validation="true" value="" type="text" hascounter="true" customclass="" required="true"></b-input-stub>
                    </b-field-stub>
                  </div>
                  <div class="column is-half">
                    <b-field-stub addons="true">
                      <b-radio-button-stub value="true" nativevalue="true" type="is-primary" expanded="true">
                        International
                      </b-radio-button-stub>
                      <b-radio-button-stub value="true" type="is-primary" expanded="true">
                        National
                      </b-radio-button-stub>
                    </b-field-stub>
                  </div>
                </div>
              </div>
              <div class="field">
                <b-field-stub label="Place of the tournament" addons="true" labelposition="on-border">
                  <b-input-stub expanded="true" icon="city" usehtml5validation="true" value="" type="text" hascounter="true" customclass="" required="true" placeholder="City"></b-input-stub>
                  <b-autocomplete-stub expanded="true" icon="earth" usehtml5validation="true" value="" data="" field="value" keepfirst="true" dropdownposition="auto" placeholder="Country"></b-autocomplete-stub>
                </b-field-stub>
              </div>
              <div class="field">
                <b-field-stub label="Dates" addons="true" labelposition="on-border">
                  <b-datepicker-stub icon="calendar-today" usehtml5validation="true" value="" daynames="Su,M,Tu,W,Th,F,S" monthnames="January,February,March,April,May,June,July,August,September,October,November,December" firstdayofweek="1" inline="true" mindate="Tue Jan 01 2019 02:00:00 GMT+0200 (Eastern European Standard Time)" maxdate="Fri Dec 31 2021 02:00:00 GMT+0200 (Eastern European Standard Time)" placeholder="Pick tournament dates" dateformatter="function _default(date, vm) {
        if (typeof __chunk_2.config.defaultDateFormatter === 'function') {
          return __chunk_2.config.defaultDateFormatter(date);
        } else {
          return defaultDateFormatter(date, vm);
        }
      }" dateparser="function _default(date, vm) {
        if (typeof __chunk_2.config.defaultDateParser === 'function') {
          return __chunk_2.config.defaultDateParser(date);
        } else {
          return defaultDateParser(date, vm);
        }
      }" datecreator="function _default() {
        if (typeof __chunk_2.config.defaultDateCreator === 'function') {
          return __chunk_2.config.defaultDateCreator();
        } else {
          return new Date();
        }
      }" mobilenative="true" indicators="dots" iconprev="chevron-left" iconnext="chevron-right" yearsrange="-100,3" nearbymonthdays="true" rulesforfirstweek="4" range="true" closeonclick="true" mobilemodal="true" focusable="true" trapfocus="true" required="true"></b-datepicker-stub>
                </b-field-stub>
              </div>
              <div class="field"><label class="label">
                  <b-button-stub type="is-text" iconright="menu-up" nativetype="button" tag="button">Notes (0)
                  </b-button-stub>
                </label>
                <!---->
                <b-field-stub label="Add a note" addons="true" labelposition="on-border">
                  <b-input-stub expanded="true" maxlength="256" usehtml5validation="true" value="" type="textarea" hascounter="true" customclass="" placeholder="Type a message" rows="1"></b-input-stub>
                  <b-button-stub type="is-info" iconleft="comment-text-outline" nativetype="button" tag="button">Add note
                  </b-button-stub>
                </b-field-stub>
                <!---->
              </div>
            </div>
            <div class="column is-half">
              <div id="td-field" class="field">
                <b-field-stub label="Tournament Director / TD" addons="true">
                  <div class="columns">
                    <div class="column is-full">
                      <b-field-stub addons="true">
                        <b-autocomplete-stub expanded="true" icon="magnify" usehtml5validation="true" value="" data="" field="value" keepfirst="true" dropdownposition="auto" placeholder="Search by name or email">
                          &gt;
                          <template></template></b-autocomplete-stub>
                        <b-tooltip-stub active="true" type="is-info" label="Set yourself as the TD" position="is-top" size="is-medium">
                          <b-button-stub type="is-info" iconleft="account-circle-outline" nativetype="button" tag="button" title="Set yourself as TD"></b-button-stub>
                        </b-tooltip-stub>
                      </b-field-stub>
                    </div>
                    <!---->
                  </div>
                </b-field-stub>
              </div>
              <div id="referees-field" class="field">
                <b-field-stub label="Referees" addons="true"></b-field-stub>
                <referee-table-stub editableitem="all" editable="true" value=""></referee-table-stub>
                <b-field-stub label="Add referee" addons="true" labelposition="on-border">
                  <b-autocomplete-stub expanded="true" icon="magnify" usehtml5validation="true" value="" data="" field="value" keepfirst="true" clearonselect="true" dropdownposition="auto" placeholder="Search by name, email or nationality"> <template></template></b-autocomplete-stub>
                  <b-tooltip-stub active="true" type="is-info" label="Add new referee to system" position="is-top" size="is-medium">
                    <b-button-stub type="is-info" iconleft="account-plus" outlined="true" nativetype="button" tag="button">
                      New referee
                    </b-button-stub>
                  </b-tooltip-stub>
                  <b-tooltip-stub active="true" type="is-info" label="Add yourself as a referee" position="is-top" size="is-medium">
                    <b-button-stub type="is-info" iconleft="account-circle-outline" nativetype="button" tag="button"></b-button-stub>
                  </b-tooltip-stub>
                </b-field-stub>
              </div>
              <div class="field">
                <b-field-stub label="Teams competing" addons="true">
                  <div>
                    <b-field-stub addons="true">
                      <b-autocomplete-stub expanded="true" usehtml5validation="true" value="" data="" field="value" keepfirst="true" clearonselect="true" dropdownposition="auto" placeholder="Team name / Country"></b-autocomplete-stub>
                      <b-button-stub type="is-info" iconleft="plus" nativetype="button" tag="button" disabled="true">
                        Add
                      </b-button-stub>
                    </b-field-stub>
                    <div class="team-list"></div>
                  </div>
                </b-field-stub>
              </div>
            </div>
          </div>
          <!---->
          <!---->
          <!---->
        </div>
        <div class="card-footer">
          <b-button-stub type="is-primary" iconleft="check-circle-outline" outlined="true" nativetype="button" tag="button" disabled="true" class="card-footer-item">
            Save
          </b-button-stub>
          <b-button-stub type="is-light" iconleft="cancel" nativetype="button" tag="button" class="card-footer-item">
            Cancel
          </b-button-stub>
          <!---->
        </div>
        <!---->
      </b-collapse-stub>
      <b-field-stub addons="true">
        <b-radio-button-stub value="own" nativevalue="own" type="is-info">
          <b-icon-stub icon="account-circle-outline"></b-icon-stub> <span>Your tournaments</span>
        </b-radio-button-stub>
        <b-radio-button-stub value="own" nativevalue="all" type="is-info">
          <b-icon-stub icon="earth"></b-icon-stub> <span>All tournaments</span>
        </b-radio-button-stub>
      </b-field-stub>
      <div class="tournament-list">
        <b-table-stub data="[object Object]" columns="" striped="true" hoverable="true" detailed="true" headercheckable="true" checkboxposition="left" isrowselectable="function _default() {
        return true;
      }" isrowcheckable="function _default() {
        return true;
      }" checkedrows="" mobilecards="true" defaultsort="name" defaultsortdirection="asc" sorticon="arrow-up" sorticonsize="is-small" sortmultipledata="" currentpage="1" perpage="20" showdetailicon="true" paginationposition="bottom" rowclass="function _default() {
        return '';
      }" openeddetailed="" hasdetailedvisible="function _default() {
        return true;
      }" detailkey="id" total="0" filtersevent=""> <template></template></b-table-stub>
      </div>
      <router-link-stub to="[object Object]" target="_blank" class="spaced">
        <b-button-stub iconright="printer" nativetype="button" tag="button">Print E-passport</b-button-stub>
      </router-link-stub>
      <!---->
    </div>
  </section>
  <section class="footer">
    <footer class="footer">
      <div class="content has-text-centered">
        <p>
          <router-link-stub to="[object Object]">About</router-link-stub>
          |
          <router-link-stub to="[object Object]">Privacy Policy</router-link-stub>
        </p>
        <p><strong>ePass</strong> by <a href="https://github.com/JSomerstone">JSomerstone</a>
          |
          The source code is licensed under <a href="http://www.apache.org/licenses/">Apache Licence v. 2</a>.
          |
          <a href="/versions.txt">Version history</a></p>
      </div>
    </footer>
  </section>
</div>
`;

exports[`Tournaments-view renders correctly with empty list 1`] = `
<layout-stub>
  <tournament-form-stub editable="true" tournament="[object Object]" year="2020" class="tournament-form"></tournament-form-stub>
  <tournament-filter-stub></tournament-filter-stub>
  <tournament-list-stub items="[object Object]" year="2020"></tournament-list-stub>
  <router-link-stub to="[object Object]" target="_blank" class="spaced">
    <b-button-stub iconright="printer" nativetype="button" tag="button">Print E-passport</b-button-stub>
  </router-link-stub>
  <!---->
</layout-stub>
`;
